### PR TITLE
Add index store to a dedicated CppCompilationOutputs field and output group

### PR DESCRIPTION
**Pull request by AlessandroPatti**
Add the index store files to a dedicated output group for an easier retrieval in Starlark.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java [hunk 1]
```diff
@@ -800,6 +800,11 @@ public CompilationInfo compile() throws RuleErrorException {
     outputGroupsBuilder.putAll(
         CcCommon.createSaveFeatureStateArtifacts(
             cppConfiguration, featureConfiguration, ruleContext));
+    outputGroupsBuilder.put(
+        "index_store",
```
Comment (rmaz) [inline src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java]: `objc_index_store` ?

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java [hunk 2]
```diff
@@ -800,6 +800,11 @@ public CompilationInfo compile() throws RuleErrorException {
...
+        NestedSetBuilder.<Artifact>stableOrder()
+            .addAll(ccCompilationOutputs.getIndexStoreFiles()).build()
```
Comment (rmaz) [inline src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java:806]: what if there aren't any?

Comment (AlessandroPatti) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java:806]: Then the output group is an empty depset

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java (hunk 1 above)
Comment (AlessandroPatti) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java]: Changed to `cc_index_store` as this is also valid for cc rules

**Review by rmaz — APPROVED** [review]

**Pull request merged**